### PR TITLE
storeapi: handle 500 error response when releasing snap

### PR DESCRIPTION
**Pull request by kyrofa**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----

This PR fixes LP: [#1750177](https://bugs.launchpad.net/snapstore/+bug/1750177) and #2028 by properly handling the store returning a 500 error formatted as HTML instead of a typical error response. It also starts printing store error responses when in debug mode in order to streamline such fixes in the future.

### CODE REVIEW

File: snapcraft/storeapi/errors.py [hunk 1]
```diff
@@ -396,6 +410,16 @@ def __fmt_error_401_or_403(self, response):
     def __fmt_error_404(self, response):
         return self.__FMT_NOT_REGISTERED
 
+    def __fmt_error_500(self, response):
+        response_json = self.__to_json(response)
+        message = ('The store encountered an internal error. Please report a '
+                   'bug if this continues.')
```
Comment (sergiusens) [inline snapcraft/storeapi/errors.py]: where?

Comment (sergiusens) [reply snapcraft/storeapi/errors.py]: also reporting a bug without much information or the fact that it is repeatable would be a really hard to fix thing. This should probably be something done with sentry.
I would state that if the error persists, to log a bug or reach out for support, but not on the first try (we could also tell them to look at the status page)

Comment (kyrofa) [reply snapcraft/storeapi/errors.py]: Alright, done.

**Comment by kyrofa** [comment]
> Also, if we are passing in response, I would remove the need to pass thing such as response.status_code.

As far as I can tell, none of the errors require this (although some format strings include them). Can you please clarify?

**Pull request merged**